From: Daniel Molkentin <daniel@molkentin.de>
Date: Tue, 21 Mar 2017 03:26:05 +0100
Subject: units: apply plymouth warning fix to in rescue mode as well (#5615)

Follow up for #5528.
(cherry picked from commit 03bf096ba283bfcba0725375f152a823e998cdbc)
---
 units/rescue.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/units/rescue.service.in b/units/rescue.service.in
index 5feff69..dc6d14b 100644
--- a/units/rescue.service.in
+++ b/units/rescue.service.in
@@ -16,7 +16,7 @@ Before=shutdown.target
 [Service]
 Environment=HOME=/root
 WorkingDirectory=-/root
-ExecStartPre=-/bin/plymouth --wait quit
+ExecStartPre=-/bin/sh -c "[ -x /bin/plymouth ] && /bin/plymouth --wait quit"
 ExecStartPre=-/bin/echo -e 'You are in rescue mode. After logging in, type "journalctl -xb" to view\\nsystem logs, "systemctl reboot" to reboot, "systemctl default" or ^D to\\nboot into default mode.'
 ExecStart=-/bin/sh -c "@SULOGIN@; @SYSTEMCTL@ --job-mode=fail --no-block default"
 Type=idle
